fvm/evm/stdlib/contract.cdc

```diff
@@ -488,4 +543,59 @@ contract EVM {
     fun getLatestBlock(): EVMBlock {
         return InternalEVM.getLatestBlock() as! EVMBlock
     }
+
+    /// Interface for a resource which acts as an entrypoint to the VM bridge
+    access(all)
+    resource interface BridgeAccessor {
+
+        /// Endpoint enabling the bridging of an NFT to EVM
+        access(all)
+        fun depositNFT(
+            nft: @NonFungibleToken.NFT,
+            to: EVMAddress,
+            feeProvider: &{FungibleToken.Provider}
+        )
+
+        /// Endpoint enabling the bridging of an NFT from EVM
+        access(all)
+        fun withdrawNFT(
+            caller: &CadenceOwnedAccount,
+            type: Type,
+            id: UInt256,
+            feeProvider: &{FungibleToken.Provider}
+        ): @NonFungibleToken.NFT
+
+        /// Endpoint enabling the bridging of a fungible token vault to EVM
+        access(all)
+        fun depositTokens(
+            vault: @FungibleToken.Vault,
+            to: EVMAddress,
+            feeProvider: &{FungibleToken.Provider}
+        )
+
+        /// Endpoint enabling the bridging of fungible tokens from EVM
+        access(all)
+        fun withdrawTokens(
+            caller: &CadenceOwnedAccount,
+            type: Type,
+            amount: UInt256,
+            feeProvider: &{FungibleToken.Provider}
+        ): @FungibleToken.Vault
+    }
+
+    /// Interface which captures a Capability to the bridge Accessor, saving it within the BridgeRouter resource
+    access(all)
+    resource interface BridgeRouter {
+
+        /// Returns a reference to the BridgeAccessor designated for internal bridge requests
+        access(all) view fun borrowBridgeAccessor(): &{BridgeAccessor}
+    }
+
+    /// Returns a reference to the BridgeAccessor designated for internal bridge requests
+    access(self)
+    view fun borrowBridgeAccessor(): &{BridgeAccessor} {
+        return self.account.borrow<&{BridgeRouter}>(from: /storage/evmBridgeRouter)
+            ?.borrowBridgeAccessor()
+            ?? panic("Could not borrow reference to the EVM bridge")
```

Comment:
do we have to use a borrow approach here, instead of keeping the bridge accessor as a field in the EVM and directly using it?

Reply:
Hmm, what do you mean? Adding a field to the EVM contract wouldn't be an upgradable change, so not sure of the specifics required for that update. Or are you suggesting something else?

Reply:
Gio is smart, this is the best way to do it at the moment.